### ISSUE TITLE
Make demobikes create an explosion big and powerful enough to destroy other demobikes in close range on death.

### ISSUE THREAD

**Issue by xamorish** [TheSuperHackers/GeneralsGamePatch#138]
This will force the GLA player to not stack demobikes and will give the opposing player's units time to take out bikes one by one and not in large groups, like it is with bomb trucks.

**Comment by xezon** [comment]
This is a design and balancing question. It certainly is reasonable from a design standpoint to apply some friendly damage. It could behave similar to Nuklear Battlemaster for example. But it changes gameplay expectations.

**Comment by xamorish** [comment]
> 
> 
> This is a design and balancing question. It certainly is reasonable from a design standpoint to apply some friendly damage. It could behave similar to Nuklear Battlemaster for example. But it changes gameplay expectations.

True. It should be decided whether this change in gameplay expectations is for the good. My personal opinion is nerfing demobikes this way should exclude the situation where the question "will i lose or not?" could be also interpreted as "will the  enemy GLA player build enough demobikes or not?".

**Comment by xezon** [comment]
I just tested some demo bikes and they do already apply slash damage to friendly units nearby.

**Comment by commy2** [comment]
But not to other Demo Bikes of the same faction, i.e. other Demo General Bikes.

See:
[before](https://github.com/xezon/GeneralsGamePatch/blob/main/Patch104pZH/GameFilesOriginal/Data/INI/Weapon.ini#L2299)
and
[after](https://github.com/xezon/GeneralsGamePatch/blob/main/Patch104pZH/GameFilesOriginal/Data/INI/Weapon.ini#L6751-L6773)
Demolitions upgrade.

The NOT_SIMILAR flag.

**Comment by xezon** [comment]
I do not understand. Meaning it applies no damage to nearby Worker on Bike for example?

**Comment by MTKing4** [comment]
This will render demo bikes useless, bomb trucks can take buildings on their own so it makes sense thay they explode in a chain-reaction manner so player can't spam them, demo bikes however can't, they function like a terrorist would not killing themselves in TTs except toxin ones

I'm against this

**Comment by commy2** [comment]
> I do not understand. Meaning it applies no damage to nearby Worker on Bike for example?

Before the Demolitions upgrade, the Demo General's Demo Bike uses the `SuicideBikeBomb` weapon when it killed itself.

`SuicideBikeBomb` affects uses the following `RadiusDamageAffects` flags:
```
SELF
SUICIDE
ALLIES
ENEMIES
NEUTRALS
NOT_SIMILAR
```
SELF and SUICIDE are pointless, since the unit is already dead.
ALLIES means the weapon does damage the players own units. However, it also uses the NOT_SIMILAR flag. This means that other Demo General Demo Bikes are not affected by the explosion. This is the case no matter what unit is the rider. However, it would still damage Bikes from other sub-factions, as these are different units.

**Comment by commy2** [comment]
After the Demolitions upgrade, the Demo General's Demo Bike *with a Terrorist on top only* uses the `Demo_TerroristOnCombatBikeSuicideDynamitePack` when it killed itself, and `Demo_TerroristOnCombatBikeSuicideDynamitePackPlusFire` when it was killed by something else.

Both of these weapons have `ALLIES` commented out. Therefore, after the Demolitions upgrade, the Demo General's Demo Bike with a Terrorist on top will not damage the players own units anymore.

**Comment by commy2** [comment]
As well after the Demolitions upgrade: If the Demo General's Demo Bike with anything else than a Terrorist on top is destroyed it will use the `Demo_SuicideDynamitePackPlusFire` weapon.

If it's suicided by ability it instead uses `Demo_SuicideDynamitePack`, however, that button is bugged and may not be available every time depending on when you put the unit on top of the Bike and when you researched the upgrade.

Both of these also have `ALLIES` commented out, thus behave the same as the Terrorist **after the Demolitons upgrade**, but those weapons deal slightly less damage with different radii etc.

**Comment by commy2** [comment]
To summarize:

Before Demolitions:
Teror Bike damages allies, except other Demo General Bikes (no matter what rider). (ALLIES + NOT_SIMILAR)

After Demolitions
Terror Bike does not damage allies. (ALLIES commented out)

**Comment by xezon** [comment]
Adding Bike Self Damage post Demo Upgrade would therefore weaken Bike effectiveness a bit. Demo player would need to be more careful when attacking with Bikes to not kill allies. Would that be desirable?

**Comment by ExiLe-ZH** [comment]
If the friendly blast radius isn’t too large (triggering chainreactions too easy), then this could maybe help nerfing db’s to more grounded levels, where more micro is needed (splitting, timing and sending from different angles) instead of simply waypointing unstoppable blobs with a few clicks.

Still very controversial but atleast worth testing imo.

**Comment by commy2** [comment]
> Would that be desirable?

It would be very confusing if they were to damage themselves only after the upgrade. If self damage is chosen, the same should happen to non-Demolitions upgrade Terrorists, which includes the other factions.

**Comment by FinAminToastCrunch** [comment]
Just make demobikes more expensive. Making them do more friendly fire makes them useless.

**Comment by ExiLe-ZH** [comment]
Revisiting this, I really like the idea of making them chain-explode easier like Nuke Battlemasters. Cause singular or a few DB's are stoppable, however gangbangs can't be stopped AT ALL and is EXACTLY what makes DB's OP. 

Wouldn't even need to alter the damage, radius or cost/production values with this singular change I don't think.

**Comment by commy2** [comment]
@ImTimK What change do you mean? It's not obvious to me how to proceed here.

**Comment by xezon** [comment]
Are you asking @ImTimK ?

**Comment by ExiLe-ZH** [comment]
> @ImTimK What change do you mean? It's not obvious to me how to proceed here.

Currently, if you kill 1 DB in a large group, the rest survive.

So if they would chain react like Nuke BM's, then you can't sent big gangs mindlessly anymore and you would have to spread them out and coordinate more.

**Comment by xezon** [comment]
Regarding Demo Bikes, I would just like to note that we have merged fix

* #2362

Which makes Terrorists and Demo Bikes (before Demo Upgrade) better vs USA.
